src/lib.rs

```diff
@@ -1,19 +1,72 @@
+// The doc_cfg feature is only available in nightly. It lets us mark items in documentation as
+// dependent on specific features.
+#![cfg_attr(docsrs, feature(doc_cfg))]
 #![no_std]
-
+#![doc = include_str!("../README.md")]
 extern crate alloc;
 
 #[cfg(feature = "std")]
 extern crate std;
 
-pub mod consistency;
-pub mod error;
-pub mod inclusion;
-mod leaf;
-mod merkle_tree;
-mod tree_math;
+pub use digest;
+
+mod consistency;
+mod error;
+mod inclusion;
+pub mod mem_backed_tree;
+mod tree_util;
 
 #[cfg(test)]
 mod test_util;
 
-pub use leaf::*;
-pub use merkle_tree::*;
+pub use consistency::*;
+pub use error::*;
+pub use inclusion::*;
+pub use tree_util::*;
+
+use digest::Digest;
+use subtle::ConstantTimeEq;
+
+/// The root hash of a Merkle tree. This uniquely represents the tree.
+#[derive(Clone, Debug)]
+pub struct RootHash<H: Digest> {
+    /// The root hash of the Merkle tree that this root represents
+    pub(crate) root_hash: digest::Output<H>,
```

Comment:
Do these fields need to even be `pub(crate)` given that we can set and access the fields using the struct's methods?
```suggestion
    root_hash: digest::Output<H>,
```